test/src/main/java/org/apache/accumulo/test/fate/FatePoolsWatcherIT.java

```diff
@@ -357,6 +357,151 @@ protected void testIdleCountHistory(FateStore<PoolResizeTestEnv> store, ServerCo
+
+    final boolean isUserStore = store.type() == FateInstanceType.USER;
+
+    final Fate.FateOperation fateOpFromPool1 =
+        isUserStore ? userPool1.iterator().next() : metaPool1.iterator().next();
+    final Fate.FateOperation fateOpFromPool2 =
+        isUserStore ? userPool2.iterator().next() : metaPool2.iterator().next();
+    final Fate.FateOperation fateOpFromPool3 =
+        isUserStore ? userPool3.iterator().next() : metaPool3.iterator().next();
+    final Fate.FateOperation fateOpFromPool4 =
+        isUserStore ? userPool4.iterator().next() : metaPool4.iterator().next();
+
+    final Set<Fate.FateOperation> pool1 = isUserStore ? userPool1 : metaPool1;
+    final Set<Fate.FateOperation> pool2 = isUserStore ? userPool2 : metaPool2;
+    final Set<Fate.FateOperation> pool3 = isUserStore ? userPool3 : metaPool3;
+    final Set<Fate.FateOperation> pool4 = isUserStore ? userPool4 : metaPool4;
+
+    boolean allAssertsOccurred = false;
+    final var env = new PoolResizeTestEnv();
+    final Fate<PoolResizeTestEnv> fate = new FastFate<>(env, store, false, r -> r + "", config);
+
+    try {
+      // seeding pool1/FateExecutor1
+      for (int i = 0; i < numSeedPool1; i++) {
+        fate.seedTransaction(fateOpFromPool1, fate.startTransaction(), new PoolResizeTestRepo(),
+            true, "testing");
+      }
+      // seeding pool2/FateExecutor2
+      for (int i = 0; i < numSeedPool2; i++) {
+        fate.seedTransaction(fateOpFromPool2, fate.startTransaction(), new PoolResizeTestRepo(),
+            true, "testing");
+      }
+      // seeding pool3/FateExecutor3
+      for (int i = 0; i < numSeedPool3; i++) {
+        fate.seedTransaction(fateOpFromPool3, fate.startTransaction(), new PoolResizeTestRepo(),
+            true, "testing");
+      }
+      // seeding pool4/FateExecutor4
+      for (int i = 0; i < numSeedPool4; i++) {
+        fate.seedTransaction(fateOpFromPool4, fate.startTransaction(), new PoolResizeTestRepo(),
+            true, "testing");
+      }
+
+      Wait.waitFor(() -> env.numWorkers.get()
+          == Math.min(numThreadsPool1, numSeedPool1) + Math.min(numThreadsPool2, numSeedPool2)
+              + Math.min(numThreadsPool3, numSeedPool3) + Math.min(numThreadsPool4, numSeedPool4));
+      // wait for all transaction runners to be active
+      Wait.waitFor(() -> fate.getTotalTxRunnersActive()
+          == numThreadsPool1 + numThreadsPool2 + numThreadsPool3 + numThreadsPool4);
+      assertEquals(numThreadsPool1, fate.getTxRunnersActive(pool1));
+      assertEquals(numThreadsPool2, fate.getTxRunnersActive(pool2));
+      assertEquals(numThreadsPool3, fate.getTxRunnersActive(pool3));
+      assertEquals(numThreadsPool4, fate.getTxRunnersActive(pool4));
+
+      // wait a bit longer to ensure another iteration of the pool watcher check doesn't change
+      // anything
+      Thread.sleep(fate.getPoolWatcherDelay().toMillis() + 1_000);
+
+      assertEquals(env.numWorkers.get(),
```

Comment:
This is a good check that verifies things from the actualy running thread perspective, however it only verifies the sum.  In addition to this check maybe could do something like the following to verifiy what is in the store at a more granular level, if it would work.   The checks together would give a stronger verification.

```
      Map<Fate.FateOperation,Long> seenCounts =  store.list().filter(fateIdStatus -> fateIdStatus.getFateOperation().isPresent() && fateIdStatus.getFateReservation().isPresent()).collect(Collectors.groupingBy(fis->fis.getFateOperation().orElseThrow(), Collectors.counting()));
      Map<Fate.FateOperation,Long> expectedCounts = Map.of(fateOpFromPool1, (long)Math.min(numThreadsPool1, numSeedPool1),...);
      assertEquals(expectedCounts, seenCounts);
```